### PR TITLE
wallet: Change ScriptPubKeyMan::Upgrade default to True

### LINKED ISSUE #22460: `upgradewallet` on a descriptor wallet ends with error
This is the error I get:

```
$ bitcoin-cli -rpcwallet=descriptor -testnet upgradewallet                         <
error code: -1
error message:
wallet/rpcwallet.cpp:4522 (operator())
Internal bug detected: '!error.empty()'
You may report this issue here: https://github.com/bitcoin/bitcoin/issues
```

The daemon logs this:

```
2021-07-15T16:16:54Z [descriptor] Performing wallet upgrade to 169900
```

**Expected behavior**

When I do the same on the default (non-descriptor) wallet:

```
$ bitcoin-cli -rpcwallet="" -testnet upgradewallet                             
{
  "wallet_name": "",
  "previous_version": 169900,
  "current_version": 169900,
  "result": "Already at latest version. Wallet version unchanged."
}
```

**System information**

Bitcoin Core v0.21.1 Linux x86_64 glibc (Ubuntu) using binarias from bitcoincore.org

### PR DESCRIPTION
**Pull request by achow101**
When adding a new ScriptPubKeyMan, it's likely that there will be nothing for `Upgrade` to do. If it is called (via `upgradewallet`), then it should do nothing, successfully. This PR changes the default `ScriptPubKeyMan::Upgrade` function so that it returns a success instead of failure when doing nothing.

Fixes #22460

### CODE REVIEW

**Comment by GeneFerneau** [comment]
Concept + review ACK [4ff4dc0](https://github.com/bitcoin/bitcoin/pull/22461/commits/4ff4dc06992eb821fb97f23d7883859924bd9c7d)

File: test/functional/wallet_upgradewallet.py [hunk 1]
```diff
@@ -352,6 +353,11 @@ def copy_split_hd():
         v16_3_kvs = dump_bdb_kv(v16_3_wallet)
         assert b'\x0adefaultkey' not in v16_3_kvs
 
+        if self.is_sqlite_compiled():
+            self.log.info("Checking that descriptor wallet do nothing, successfully")
```
Comment (meshcollider) [inline test/functional/wallet_upgradewallet.py:357]: Nit: "descriptor wallet do nothing"

Comment (achow101) [reply test/functional/wallet_upgradewallet.py:357]: Fixed

**Review by jonatack — COMMENTED** [review]
ACK 5012a7912ee9fa35bc417cb073eebffd85f36c6c

File: test/functional/wallet_upgradewallet.py (hunk 1 above)
Comment (jonatack) [reply test/functional/wallet_upgradewallet.py]: Doing nothing successfully sounds very nice :)

**Review by meshcollider — COMMENTED** [review]
utACK 5012a7912ee9fa35bc417cb073eebffd85f36c6c

**Pull request merged**